subprojects/core-api/src/main/java/org/gradle/api/artifacts/transform/TransformParameters.java

```diff
@@ -21,13 +21,30 @@
 /**
  * Marker interface for parameter objects to {@link TransformAction}s.
  *
+ * <p>
+ *     Parameter types should use be interfaces, only declaring getters for {@link org.gradle.api.provider.Property} like objects.
+ *     All getters need to be annotated one of the input annotations {@link org.gradle.api.tasks.Console}, {@link org.gradle.api.tasks.Input}, {@link org.gradle.api.tasks.InputDirectory}, {@link org.gradle.api.tasks.InputFile}, {@link org.gradle.api.tasks.InputFiles}, {@link org.gradle.api.tasks.Internal} or {@link org.gradle.api.model.ReplacedBy}.
+ *     Normalization annotations such as {@link org.gradle.api.tasks.PathSensitive} or {@link org.gradle.api.tasks.Classpath} can be used as well.
+ *     Example:
+ * </p>
+ * <pre>
+ * interface MyParameters extends TransformParameters {
+ *     {@literal @}Input
+ *     Property&lt;String&gt; getStringParameter();
+ *     {@literal @}InputFiles
+ *     ConfigurableFileCollection getInputFiles();
+ * }
+ * </pre>
```

Comment:
I think these can be auto-tested already, no?

Reply:
They are auto-tested now.

Reply:
How? I only see `<pre>` here.

Reply:
Fixed now.

Reply:
How does that work, so I know for future reference?

Reply:
@pledbrook You need to add `class='autoTested'` to the `<pre>` tag. There are some tests which pick up those tags in the Javadoc and run a build with the code snippet. For example, this is the test for everything in `coreApi`: https://github.com/gradle/gradle/blob/998f9b7a1d9c0b8cda779eae213f466b6fa23024/subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/samples/AutoTestedSamplesCoreApiIntegrationTest.groovy#L22-L33